app/derivative-services/jp2_derivative_service.rb

```diff
@@ -114,7 +120,28 @@ def use
   end
 
   def filename
-    return Pathname.new(file_object.io.path) if file_object.io.respond_to?(:path) && File.exist?(file_object.io.path)
+    Pathname.new(file_object.disk_path)
+  end
+
+  def clean_filename
+    @filename ||=
+      begin
+        Pathname.new(cleaned_file.path)
+      end
+  end
+
+  # Remove compression from given TIFF file, just in case.
+  def cleaned_file
+    @cleaned_file ||=
+      begin
+        t = Tempfile.new
+        MiniMagick::Tool::Convert.new do |convert|
+          convert << file_object.disk_path.to_s
+          convert.compress.+
```

Comment:
what does this do? Is it some kind of DSL for the command line option?

Reply:
Yeah. Makes the command line `compress+`